packages/app-check/src/internal-api.test.ts

```diff
@@ -197,35 +196,48 @@ describe('internal api', () => {
       expect(errorFn1.args[0][0].name).to.include('exchange error');
     });
 
+    it('ignores listeners that throw', async () => {
+      activate(app, FAKE_SITE_KEY, false);
+      stub(reCAPTCHA, 'getToken').resolves(fakeRecaptchaToken);
+      stub(client, 'exchangeToken').resolves(fakeRecaptchaAppCheckToken);
+      const listener1 = (): void => {
+        throw new Error();
+      };
+      const listener2 = spy();
+
+      const errorFn1 = spy();
+
+      addTokenListener(app, fakePlatformLoggingProvider, listener1, errorFn1);
+      addTokenListener(app, fakePlatformLoggingProvider, listener2);
+
+      await getToken(app, fakePlatformLoggingProvider);
+
+      expect(errorFn1).not.to.be.called;
+      expect(listener2).to.be.called;
+    });
+
     it('loads persisted token to memory and returns it', async () => {
-      const clock = useFakeTimers();
       activate(app, FAKE_SITE_KEY);
 
-      stub(storage, 'readTokenFromStorage').returns(
-        Promise.resolve(fakeCachedAppCheckToken)
-      );
+      storageReadStub.resolves(fakeCachedAppCheckToken);
 
       const clientStub = stub(client, 'exchangeToken');
 
       expect(getState(app).token).to.equal(undefined);
-      expect(await getToken(app, fakePlatformLoggingProvider)).to.deep.equal({
+      const result = await getToken(app, fakePlatformLoggingProvider);
+      expect(result).to.deep.equal({
         token: fakeCachedAppCheckToken.token
       });
       expect(getState(app).token).to.equal(fakeCachedAppCheckToken);
       expect(clientStub).has.not.been.called;
-
-      clock.restore();
     });
 
     it('persists token to storage', async () => {
-      activate(app, FAKE_SITE_KEY);
+      activate(app, FAKE_SITE_KEY, false);
 
-      stub(storage, 'readTokenFromStorage').returns(Promise.resolve(undefined));
-      stub(reCAPTCHA, 'getToken').returns(Promise.resolve(fakeRecaptchaToken));
-      stub(client, 'exchangeToken').returns(
-        Promise.resolve(fakeRecaptchaAppCheckToken)
-      );
-      const storageWriteStub = stub(storage, 'writeTokenToStorage');
+      stub(reCAPTCHA, 'getToken').resolves(fakeRecaptchaToken);
+      stub(client, 'exchangeToken').resolves(fakeRecaptchaAppCheckToken);
+      storageWriteStub.resetHistory();
```

Comment:
what does resetHistory do?

Reply:
If the stub persists from test to test, it clears the call history so "calledOnce" is true even if it's called every time, but considering the stubs are all cleared and re-made after each test it's not needed here. Removed.